packages/widget/src/pages/SettingsPage/SlippageSettings/SlippageSettings.tsx

```diff
@@ -24,36 +22,48 @@ export const SlippageSettings: React.FC = () => {
   const { isSlippageOutsideRecommendedLimits, isSlippageChanged } =
     useSettingMonitor();
   const { slippage } = useSettings(['slippage']);
-  const setValue = useSettingsStore((state) => state.setValue);
+  const { setValue } = useSettingsActions();
   const defaultValue = useRef(slippage);
   const [focused, setFocused] = useState<'input' | 'button'>();
 
+  const customInputValue =
+    !slippage || slippage === defaultSlippage ? '' : slippage;
+
+  const [inputValue, setInputValue] = useState(customInputValue);
+
   const handleDefaultClick = () => {
     setValue('slippage', formatSlippage(defaultSlippage, defaultValue.current));
   };
 
-  const handleInputUpdate: ChangeEventHandler<HTMLInputElement> = (event) => {
-    const { value } = event.target;
+  const debouncedSetValue = useMemo(() => debounce(setValue, 1000), [setValue]);
 
-    setValue(
-      'slippage',
-      formatSlippage(value || defaultSlippage, defaultValue.current, true),
-    );
-  };
+  const handleInputUpdate: ChangeEventHandler<HTMLInputElement> = useCallback(
+    (event) => {
+      const { value } = event.target;
+
+      setInputValue(formatSlippage(value, defaultValue.current, true));
+
+      debouncedSetValue(
+        'slippage',
+        formatSlippage(value || defaultSlippage, defaultValue.current, true),
+      );
+    },
+    [debouncedSetValue],
+  );
```

Comment:
Had to decouple the inputs value from setting the slippage value in the settings store. the debounce means this slippage value happens once and not on each keystroke